### PR TITLE
Merge `math-components` with `develop` branch

### PR DESCRIPTION
**Pull request by o-rashad**
#### What's new :question: 
- New components created : `Calculator`,  `Screen` , `Keyboard`, `Key`
- React Class based components used.
- Update README file
- Refactor the codebase

#### Note :stop_sign: 
- `react/prefer-stateless-function` and `no-useless-constructor` rules have been disabled to coup with the project requirements creating components using class based syntax.

### CODE REVIEW

File: src/components/key/Key.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  }
+
+  render() {
+    const { op } = this.props;
+    const { display } = this.props;
+    return (
+      <div className={op ? 'calKey opColor' : 'calKey'}>
+        {display}
+      </div>
+    );
```
Comment (omar-labana) [inline src/components/key/Key.js:18]: - [x] Clicking any of the buttons triggers this error with no output to the result screen.
image

**Review by omar-labana — CHANGES_REQUESTED** [review]
Hi @od-c0d3r, 

Excellent work so far working with class-based components! 👌
You have implemented almost all the required features. ✔️ 
There are some cases that you need to adjust but you are almost done! 😀

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
- [x] _[OPTIONAL]_ Please re-add your build folder to `.gitignore`, I'd suggest that you create a github action or a bash script to deploy your static SPA to gh-pages.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/key/Key.js (hunk 1 above)
Comment (o-rashad) [reply src/components/key/Key.js:18]: Thanks @omar-labana for your review,

I can't locate the bug on my version! :thinking: 
Could you help me reproduce the error on my end ? 
like give me steps to show the error message.

Comment (o-rashad) [reply src/components/key/Key.js:18]: Thanks @omar-labana for your Slack feedback
i was able to reproduce the error after cloning the repo, install depends, start server.

**Review by bimbolabuari — APPROVED** [review]
# Project Approved 🚀

Hi, @od-c0d3r 🥇

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by o-rashad** [comment]
Thanks @bimbolabuari for the review 👍

**Pull request merged**